### ISSUE TITLE
Incorrect adj.P.Vals in table of glMDPlot

### ISSUE THREAD

**Issue by charitylaw** [Shians/Glimma#124]
Check rounding of adjusted p-values (when using topTreat). Zero values appear at the top of the table for adjusted p-values, when they are not present using topTreat.

**Comment by ivirshup** [comment]
I think I ran into this error (or at least a similar one). If there is any value `x` where `1e-05 <= x < 5e-5` you'd like to display in you Glimma plot, it will be shown as 0. I ran into this after I noticed some incorrect sorting of tables generated by `glMDPlot.default`. Here's an example using `glScatter.default`:

```r
library(Glimma)
library(jsonlite)
library(ggplot2)

plotdata = data.frame(x=exp(log(10)*seq(log10(1e-7),log10(1),length.out=1000)), y=1:1000)
plot = Glimma:::glScatter.default(plotdata, xval="x", yval="y")

jsondf = fromJSON(plot$json)
plotdata$ZeroVal = jsondf$x == 0
summary(plotdata[plotdata$ZeroVal == T, "x"])
```
```
Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
1.009e-05 1.505e-05 2.243e-05 2.494e-05 3.344e-05 4.985e-05 
```

Visualization:

```r
ggplot(plotdata, aes(x=x)) + 
	ggtitle("'x' by whether glimma says it's zero") + 
	geom_histogram(binwidth = .1) +
	facet_grid(ZeroVal ~ ., scales="free_y") + 
	scale_x_log10(limits=c(5e-8, 1.1))
```
![glimma-issue_rounding-error](https://cloud.githubusercontent.com/assets/8238804/26577035/4c63bc92-456e-11e7-89d2-b50b9b2b60e8.png)


I think the underlying problem is with the `jsonlite` library, (I've opened [an issue](https://github.com/jeroen/jsonlite/issues/184) there).

```r
toJSON(plotdata[1:10*100,"x"])
# [4.9396e-07,2.4797e-06,0,0.0001,0.0003,0.0016,0.0079,0.0397,0.1992,1] 
```

Until that is fixed, incorrect ordering in `Glimma` plots can largely be corrected by setting `digits=5` or greater in every `toJSON` call in `Glimma`.

```
R version 3.4.0 (2017-04-21)
Platform: x86_64-apple-darwin16.5.0 (64-bit)
Running under: macOS Sierra 10.12.5

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_2.2.1 jsonlite_1.4  Glimma_1.4.0 

loaded via a namespace (and not attached):
 [1] locfit_1.5-9.1   Rcpp_0.12.11     edgeR_3.18.1     lattice_0.20-35 
 [5] grid_3.4.0       plyr_1.8.4       gtable_0.2.0     magrittr_1.5    
 [9] scales_0.4.1     stringi_1.1.5    rlang_0.1.1      reshape2_1.4.2  
[13] lazyeval_0.2.0   limma_3.32.2     labeling_0.3     tools_3.4.0     
[17] stringr_1.2.0    munsell_0.4.3    compiler_3.4.0   colorspace_1.3-2
[21] tibble_1.3.3   
```

**Comment by Shians** [comment]
Thanks a lot for tracking down this issue! I'll probably have it patched up by next week.

**Comment by Shians** [comment]
Attempted fix with #127.

**Comment by ivirshup** [comment]
#127 didn't seem to fix the issue for me. I've got something else which I think works. I'll open a pull request with the changes.

**Comment by Shians** [comment]
Fixed in #128
